core/src/main/scala/cats/data/NonEmptyMapImpl.scala

```diff
@@ -93,6 +93,11 @@ sealed class NonEmptyMapOps[K, A](val value: NonEmptyMap[K, A]) {
    */
   def lookup(k: K): Option[A] = toSortedMap.get(k)
 
+  /**
+   * Applies f to the value stored at k. If lookup misses, does nothing.
+   */
+  def updateWith(k: K)(f: A => A): NonEmptyMap[K, A] = lookup(k).fold(value)(v => add((k, f(v))))
```

Comment:
Pattern matching on the Option allows you to avoid the allocation in the empty case of fold. The code isn’t as pretty but we’ve benchmarked before and is measurable so we generally use pattern matching over functions taking lambdas inside cats methods.

Reply:
I think I've seen a similar comment on SO before. What's the allocation for, the lambda?

Reply:
Both the lazy parameter to the empty case of fold and the lambda will allocate and neither will be inlined (in all probability) by the JVM.

By contrast neither problem exists with matching.

Reply:
In this case, the lazy parameter to fold is just a reference to the current data structure. Is it copied or something? I totally agree with you I'm just curious.